### ISSUE TITLE
Documentation showing placeholders instead of text

### ISSUE THREAD

**Issue by MicaelJarniac** [pandas-dev/pandas#39115]
#### Location of the documentation

[pandas.DataFrame.mad](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.mad.html)

#### Documentation problem

The documentation is showing placeholders instead of the actual text.
![Screenshot](https://i.imgur.com/inPHZSf.png)
`{desc}`, `{axis_descr}`, `{name1}`, `{name2}`, `{see_also}`, `{examples}`

https://github.com/pandas-dev/pandas/blob/3e89b4c4b1580aa890023fc550774e63d499da25/pandas/core/generic.py#L10890-L10901

**Comment by simonjayhawkins** [comment]
Thanks @MicaelJarniac for the report. docs rendered correctly in 1.1.5

**Comment by MicaelJarniac** [comment]
https://github.com/pandas-dev/pandas/blob/2aadaa875944301fb15eadf51726c0ea23973579/pandas/core/generic.py#L10491-L10510

I believe this _could_ be related. I'm just poking inside to see if I can find something, but I have ~absolutely~ no idea what's going on.

**Comment by MicaelJarniac** [comment]
While poking inside, I think I found some other places where this is happening:

[pandas.DataFrame.tz_convert](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.tz_convert.html)

https://github.com/pandas-dev/pandas/blob/2aadaa875944301fb15eadf51726c0ea23973579/pandas/core/generic.py#L9529-L9555

**Comment by MicaelJarniac** [comment]
The documentation for [pandas.DataFrame.shift](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.tz_convert.html) also seems to use the same kind of placeholders, but it renders correctly.

https://github.com/pandas-dev/pandas/blob/2aadaa875944301fb15eadf51726c0ea23973579/pandas/core/generic.py#L9133-L9236

It has a decorator, and I believe that might be what's missing on the other ones.

https://github.com/pandas-dev/pandas/blob/2aadaa875944301fb15eadf51726c0ea23973579/pandas/core/generic.py#L9133

**Comment by MicaelJarniac** [comment]
take

**Comment by MicaelJarniac** [comment]
I've managed to find the source of this issue and am now testing my fix.
Basically, when the `mad` method is created without the `@doc(...)` decorator, it doesn't get a `_docstring_components` attribute added to it.

Because of it not having a `_docstring_components`, when the next `mad` function is created (the one under the `_add_numeric_operations` method), it treats the docstring of the previous `mad` as a finished, plaintext, one, thus ignoring all the templating.

By adding an empty `@doc()` decorator on top of the first `mad` method, it gets the `_docstring_components` attribute added to it, so while generating the docstring for the next `mad`, it correctly parses the templating from the previous one.

There's a catch, though, and that's where my workaround comes in: simply adding an empty `@doc()` decorator will instantiate its `**params` with an empty `dict`, but since its corresponding docstring contains formatting placeholders, when the docstring gets parsed, it attempts to fill the placeholders using `**params`, and that results in an error, because it can't find the corresponding keys on the (empty) `dict`.

I've added a check on this parsing step, that first ensures that `**params` is not empty before attempting to use it for filling placeholders. If it's empty, the docstring is treated as plaintext.

By doing that, the empty `@doc()` decorator can now be used on objects with docstrings containing formatting placeholders, and that ensures they also get the `_docstring_components` attribute added.
